### PR TITLE
Sort out naming in XDMFFile methods

### PR DESCRIPTION
**Pull request by michalhabera**
Solves issue #20 . 

But there is still a thing to be discussed. New proposed name `write_vertex_values` is used also for `DG0` functions (and in future could be used also for VTKish XDMF for `P2` functions). Might be a bit misleading what vertex values for DG0 are...

### CODE REVIEW

File: python/src/io.cpp [hunk 1]
```diff
@@ -229,17 +229,17 @@ void io(py::module& m)
   // dolfin::io::XDMFFile::write
   xdmf_file
       // Function
-      .def("write",
+      .def("write_vertex_values",
```
Comment (garth-wells) [inline python/src/io.cpp]: Use the C++14 style syntax from pybind11 for overloaded functions.

File: python/test/unit/io/test_XDMF_P2.py [hunk 2]
```diff
@@ -34,4 +34,4 @@ def test_read_write_p2_function(tempdir):
```
Comment (garth-wells) [inline python/test/unit/io/test_XDMF_P2.py]: Do we need another file? Can it all go in `test_XDMF.py`?

Comment (michalhabera) [reply python/test/unit/io/test_XDMF_P2.py]: I don't know why is it separated. Guess that @chrisrichardson wanted to have P2 test alone to run them faster(?). He'll probably remove/merge tests after higher order geometry is settled.

But anyway XDMF tests file is getting a bit bloated. From clarity reasons I'd suggest to split it.

Comment (garth-wells) [reply python/test/unit/io/test_XDMF_P2.py]: XDMF test file is bloated, but the fix is to clean it up rather than create a new file.

**Comment by michalhabera** [comment]
@blechta I completely agree, the behaviour for "DG0" is the point I mentioned in PR description. Let me comment more on this:
1.) First, we want to distinguish two XDMF ways of writing a function to a file. The standard VTKish one (celldata for DG0 or pointdata for P1, P2) vs. the new FiniteElementFunction format.
2.) We want to let user know explicitly what is going on, since VTKish P2 is not yet implemented in dolfin. When you write in VTKish way P2 (or any other unsupported element) it saves just vertex values.

The solution I suggest:
Let us distinguish VTKish and FiniteElementFunction XDMF outputs on the method's name level, i.e. 
have e.g. `write_nodal_values` for VTKish and `write` for the new way. The fact that unsupported elements are interpolated to vertex values will be strongly emphasized in docs and in a warning message to user.

Agree, `XDMFFile::has_cell_centred_data()` is strange, because it is how XDMF thinks - there is this data being "centered" to cells, nodes, edges, etc. It is the XDMF specification. But let this be another issue, not related to this PR.

**Comment by blechta** [comment]
Ok, `XDMFFile::has_cell_centred_data()` can be another issue. In fact it might not be needed at all if we follow the suggestion I will state now.

Let's have `write_vertex_values()`, `write_cell_values()`, `write_p2_values()` with the following behaviour:

* `write_vertex_values()` interpolates vertex values if needed and writes vertex values,
* `write_cell_values()` interpolates cell values if needed and writes cell values,
* `write_p2_values()` interpolates P2 values if needed and writes P2 values.

**Comment by michalhabera** [comment]
You are addressing the problem 2.) I described. That problem is actually also present in VTKFile class = same cell centered data check and consequent interpolation to P1. In order to stay consistent we would have to apply `write_vertex_values, write_cell_values, write_p2_values` also in VTKFile.

@blechta The problem is more complex. There is still this sub-, iso-, and super-paramateric "elements" problem. Think of "write_vertex_values" for P1 function on P2 mesh. VTK has *only* isoparametric elements. We must save our P1 function as P2 function with interpolated values at midpoints for VTK (not implemented yet). Then user might be confused why write_vertex_values when it writes also midpoint values...

Maybe follow VTK terminology and use: 
1.) `write_cell_data` interpolates to (not necessarily isoparametric) P0 if needed and writes cell data,
2.) `write_point_data` interpolates to isoparametric P1 (or isoparametric P2 if implemented) if needed and writes. With proper warning to user if nontrivial interpolation (data loss) is happening.

We can use these names for both, VTKFile and XDMFFile with the same implementation meaning.

**Comment by blechta** [comment]
Yes, that makes sense. XDMF/VTK provides two or three modes: cell data, vertex data, and maybe P2 data. All of them isoparametric. All of them should be invoked by a separate function in DOLFIN explicitly.

**Comment by michalhabera** [comment]
Ok. I am closing this PR and updating the issue #20 .

**Pull request closed without merging**